ietf/sync/rfceditor.py

```diff
@@ -504,56 +556,96 @@ def parse_relation_list(l):
-                changes.append("added Errata tag")
-            has_verified_errata = any([ er['errata_status_code']=='Verified' for er in doc_errata ])
-            if has_verified_errata and not doc.tags.filter(pk=tag_has_verified_errata.pk).exists():
+                rfc_changes.append("added Errata tag")
+            has_verified_errata = any(
+                [er["errata_status_code"] == "Verified" for er in doc_errata]
+            )
+            if (
+                has_verified_errata
+                and not doc.tags.filter(pk=tag_has_verified_errata.pk).exists()
+            ):
                 doc.tags.add(tag_has_verified_errata)
-                changes.append("added Verified Errata tag")
+                rfc_changes.append("added Verified Errata tag")
         else:
             if doc.tags.filter(pk=tag_has_errata.pk):
                 doc.tags.remove(tag_has_errata)
                 if all_rejected:
-                    changes.append("removed Errata tag (all errata rejected)")
+                    rfc_changes.append("removed Errata tag (all errata rejected)")
                 else:
-                    changes.append("removed Errata tag")
+                    rfc_changes.append("removed Errata tag")
             if doc.tags.filter(pk=tag_has_verified_errata.pk):
                 doc.tags.remove(tag_has_verified_errata)
-                changes.append("removed Verified Errata tag")
-
-        if changes:
-            events.append(DocEvent.objects.create(
-                doc=doc,
-                rev=doc.rev,
-                by=system,
-                type="sync_from_rfc_editor",
-                desc="Received changes through RFC Editor sync (%s)" % ", ".join(changes),
-            ))
-
-            doc.save_with_history(events)
-
-        if changes:
-            yield changes, doc, rfc_published
+                rfc_changes.append("removed Verified Errata tag")
+
+        # Update the draft first
+        if draft_changes:
+            draft_events.append(
+                DocEvent.objects.create(
+                    doc=draft,
+                    rev=doc.rev,
+                    by=system,
+                    type="sync_from_rfc_editor",
+                    desc=f"Received changes through RFC Editor sync ({', '.join(draft_changes)})",
+                )
+            )
+            draft.save_with_history(draft_events)
+            yield draft_changes, draft, False  # yield changes to the draft
+
+        if rfc_changes:
+            rfc_events.append(
+                DocEvent.objects.create(
+                    doc=doc,
+                    rev=doc.rev,
+                    by=system,
+                    type="sync_from_rfc_editor",
+                    desc=f"Received changes through RFC Editor sync ({', '.join(rfc_changes)})",
+                )
+            )
+            doc.save_with_history(rfc_events)
+            yield rfc_changes, doc, rfc_published  # yield changes to the RFC
```

Comment:
will this yielding do the right thing if there are both draft_changes and rfc_changes?

Reply:
Yes:

Reply:
(this is also confirmed *in situ* by `test_rfc_index()`)

Reply:
(rather, will be when I commit that piece)